src/filters/write_vtt.c

```diff
@@ -236,15 +234,14 @@ GF_Err vttmx_process(GF_Filter *filter)
 	pck = gf_filter_pid_get_packet(ctx->ipid);
 	if (!pck) {
 		if (gf_filter_pid_is_eos(ctx->ipid)) {
-			gf_filter_pid_set_eos(ctx->opid);
-			if (ctx->first) {
+			if (!ctx->noempty) {
```

Comment:
if :!noempty is set, this will always inject an empty VTT on EOS even if some VTT cues have been sent, which is not desired. You need to combine this with a counter of packets sent

Reply:
True, I guess when I removed `ctx->first` I forgot about that